### PR TITLE
fix(desktop): bound every string the code-mode decoder draws

### LINKED ISSUE #2977: Bring the code-mode decoder onto the bounded wire readers
Follow-up to #2960. The two UI decoders now share one primitive module, `crates/tidebreak-desktop/ui/src/lib/wireDecode.ts`, but they still keep two string conventions side by side:

- `api/parsers.ts` (chat) bounds every string it will draw (`bounded`, `nonEmptyBounded`, `boundedBlock`) and rejects control and bidirectional characters, with named limits for ids, timestamps, and cursors.
- `code/parsers.ts` (code mode) checks presence only (`nonEmptyString`, `optionalString`, `nullableString`) with no length limit anywhere and no character clamp.

Moving code mode onto the bounded readers is a behavior change: it can start rejecting snapshots that today parse, so it needs a survey of which code-mode fields are rendered on one line versus as blocks, which carry model- or tool-authored text, and whether any field legitimately exceeds the chat limits (long titles, branch names, PR bodies). Decide per field, add the bounds, and cover the edge cases in `parsers.test.ts`.

The module doc in `wireDecode.ts` records the divergence until this lands.

### PR DESCRIPTION
**Pull request by naingthet**
## What changed

The code-mode decoder (`src/code/parsers.ts`) checked string presence only, so a snapshot could carry a control or bidirectional character straight into a one-line row, and no field had a length limit. Every string it keeps now goes through one of three tiers, chosen by how the field is drawn:

- **Line** (`bounded` / `nonEmptyBounded`, 4,096 chars): titles, names, branch and base refs, paths, URLs, logins, statuses, models, versions. Control and bidirectional characters are rejected. The limit holds a PATH_MAX path or a long URL; GitHub caps titles at 256 and refs well under that.
- **Block** (`boundedBlock`, 1 MiB): PR and comment bodies, commit messages, setup and archive scripts, quick-action commands, model text and reasoning, remediation and error messages, recaps, fence details. Line breaks, carriage returns, and tabs pass; everything else a line rejects is still rejected.
- **Raw** (`boundedRaw`, 4 MiB, length only): blob content, diffs and patches, terminal reads, command stdout/stderr, tool result previews, search hit lines, the harness's raw approval JSON, and the user's own prompt and steer text. These panes render the payload as it is and expect terminal escapes and carriage returns in it. The limit sits well over the server's own truncation points (512 KiB blobs, 256 KiB diffs, 32 KiB terminal reads).

Ids, timestamps, and cursors share `MAX_WIRE_ID_CHARS`, `MAX_WIRE_TIMESTAMP_CHARS`, and `MAX_WIRE_CURSOR_CHARS` with the chat decoder. Enum-like discriminators (`type`, `kind`) stay presence-only because they are matched, never drawn. The per-field decisions are recorded at the top of the string-bounds section in `parsers.ts`.

Two changes to the shared module (`src/lib/wireDecode.ts`):

- `boundedBlock` now admits `\r`. GitHub-authored bodies arrive with CRLF line endings, and a block renderer treats a carriage return as a space (CSS Text 3 §4.1.1), so rejecting the whole payload over it would blank most PR detail panes. This also reaches the chat decoder's exec-output tail, where a progress bar's `\r` was previously fatal to the row.
- The readers scan with one regex and count code points only when the UTF-16 length exceeds the limit, so a 1 MiB body no longer allocates two arrays per check. Semantics are unchanged; the existing code-point test still pins them.

New primitives: `boundedRaw` and `boundedStringList`. The presence-only readers remain for `settings/mcpImport.ts` and the discriminators. The module doc that recorded the divergence is replaced with the tier description.

## Why

Follow-up to #2960, which unified the two decoders over one primitive module but left their string conventions apart. Closes #2977.

## Verification

- `pnpm exec biome format src`, `pnpm lint`, `tsc --noEmit`, `pnpm test` (263 files, 2320 tests pass), `pnpm build`.
- New tests in `src/code/parsers.test.ts` (a field at the limit passes, one past it is rejected, control and bidi characters on a line field are rejected, CRLF and tabs in a block pass, verbatim payloads keep escapes, a user's pasted text never drops a turn) and `src/lib/wireDecode.test.ts` (`\r` in blocks, `boundedRaw`, `boundedStringList`, code-point counting past the UTF-16 fast path). Five of the six parser tests fail against the previous `parsers.ts`; the sixth pins the raw tier for user text.

### CODE REVIEW

File: crates/tidebreak-desktop/ui/src/code/parsers.ts [hunk 1]
```diff
@@ -2856,15 +2930,15 @@ export function parseCodeWorkspaceSearch(
...
+      !wireId(item.workspace_id) ||
+      !nonEmptyLine(item.workspace_title) ||
+      !wireId(item.session_id) ||
+      (item.turn_id !== undefined && !wireId(item.turn_id)) ||
       !["turn_user_input", "turn_narrative", "event"].includes(
         item.source as string,
       ) ||
-      typeof item.preview !== "string" ||
-      !nonEmptyString(item.created_at)
+      !blockText(item.preview) ||
```
[the remark this answers is not in the document]
Comment (naingthet) [reply crates/tidebreak-desktop/ui/src/code/parsers.ts]: Fixed in 5a7de6014ac86b8ff009dfafedff4701bed9819f: `history_matches[].preview` now reads through `rawText`, matching the prompt and tool-preview fields the excerpt is cut from, with a test that pins a colored test-run excerpt.
